### PR TITLE
Issue 42197: flow: metrics for imported run types

### PR DESCRIPTION
**Pull request by labkey-kevink**
#### Rationale
Add flow folder metrics to count flow data types and how the data was imported

### CODE REVIEW

**Comment by labkey-kevink** [comment]
Here's an example of the generated json

[flow-metrics.txt](https://github.com/LabKey/commonAssays/files/5820901/flow-metrics.txt)

File: flow/src/org/labkey/flow/controllers/FlowController.java [hunk 1]
```diff
@@ -412,6 +391,23 @@ public void addNavTrail(NavTree root)
         }
     }
 
+    @RequiresPermission(AdminPermission.class)
+    public class MetricsAction extends ReadOnlyApiAction<Object>
```
Comment (labkey-jeckels) [inline flow/src/org/labkey/flow/controllers/FlowController.java:395]: You may know this already, but you can get a preview of the full JSON metrics via Admin Console->Site Settings as well. No problems with having this action available too.

**Review by labkey-jeckels — APPROVED** [review]
This looks fine. Do you anticipate use cases for viewing metrics where you don't start at the root container and include everything? Doing it on a per-folder basis and summing will be slower than just a simple unfiltered count.

**Comment by labkey-kevink** [comment]
No, I don't expect to get metrics for a single container.  It was just easier to write the queries using LabKey SQL and the flow schema doesn't work with cross container queries.  If gathering the metrics is slow, I could rewrite the queries in database sql.

**Pull request merged**